tsl/src/continuous_aggs/options.c

```diff
@@ -117,11 +150,20 @@ cagg_find_groupingcols(ContinuousAgg *agg, Hypertable *mat_ht)
 	{
 		SortGroupClause *cagg_gc = (SortGroupClause *) lfirst(lc);
 		TargetEntry *cagg_tle = get_sortgroupclause_tle(cagg_gc, finalize_query->targetList);
-		/* groupby clauses are columns from the mat hypertable */
-		Assert(IsA(cagg_tle->expr, Var));
-		Var *mat_var = castNode(Var, cagg_tle->expr);
-		char *mat_colname = get_attname(mat_relid, mat_var->varattno, false);
-		retlist = lappend(retlist, mat_colname);
+
+		if (ContinuousAggIsFinalized(agg))
+		{
+			/* "resname" is the same as "mat column names" in the finalized version */
+			if (!cagg_tle->resjunk && cagg_tle->resname)
+				retlist = lappend(retlist, cagg_tle->resname);
+		}
+		else
+		{
+			/* groupby clauses are columns from the mat hypertable */
+			Assert(IsA(cagg_tle->expr, Var));
```

Comment:
this is redundant as castNode will Assert the correct type

Reply:
removed